p2p/src/network/noise/p2p_network_noise_reducer.rs

```diff
@@ -68,13 +68,15 @@ impl P2pNetworkNoiseState {
                 let mut offset = 0;
                 loop {
                     let buf = &self.buffer[offset..];
-                    if buf.len() >= 2 {
-                        let len = u16::from_be_bytes(buf[..2].try_into().expect("cannot fail"));
+                    let len = buf
+                        .get(..2)
+                        .and_then(|buf| Some(u16::from_be_bytes(buf.try_into().ok()?)));
```

Comment:
Here we silently ignore errors when converting the result to optional.
I implemented a `bug_condition!()` macro that I'll push in my next PR that can be used to either log the errors (and continue) or crash the node depending on an environment variable.
Please add TODO comment here so I can add a `bug_condition!()` call later.